src/operations/command.ts

```diff
@@ -98,15 +147,14 @@ export abstract class CommandOperation<
     // TODO: consider making this a non-enumerable property
     this.server = server;
 
-    const options = this.options;
     const serverWireVersion = maxWireVersion(server);
     const inTransaction = this.session && this.session.inTransaction();
 
     if (this.readConcern && commandSupportsReadConcern(cmd) && !inTransaction) {
       Object.assign(cmd, { readConcern: this.readConcern });
     }
 
-    if (options.collation && serverWireVersion < SUPPORTS_WRITE_CONCERN_AND_COLLATION) {
+    if (this.builtOptions.collation && serverWireVersion < SUPPORTS_WRITE_CONCERN_AND_COLLATION) {
```

Comment:
Any reason you changed these `options` to `this.builtOptions` instead of changing above to `const options = this.builtOptions`? Seems like we're calling that deepFreeze getter repeatedly without need here.

Reply:
Good catch I'm going through and changing, in fact I think I want to make this a method now, leaving this much work in a property access its easy to overlook that its doing work each time.